src/oemof/solph/flows/_non_convex_flow_block.py

```diff
@@ -469,24 +474,29 @@ def min_downtime_rule(_, i, o, t):
             """
             Rule definition for min-downtime constraints of non-convex flows.
             """
-            if self._time_step_allows_flexibility(
-                t, m.flows[i, o].nonconvex.max_up_down, m.TIMESTEPS.at(-1)
+            if (
+                m.flows[i, o].nonconvex.first_flexible_timestep
+                < t
+                < m.TIMESTEPS.at(-1)
             ):
                 expr = 0
                 expr += (
                     self.status[i, o, t - 1] - self.status[i, o, t]
                 ) * m.flows[i, o].nonconvex.minimum_downtime
                 expr += -m.flows[i, o].nonconvex.minimum_downtime
                 expr += sum(
-                    self.status[i, o, t + d]
-                    for d in range(0, m.flows[i, o].nonconvex.minimum_downtime)
+                    self.status[i, o, d]
+                    for d in range(
+                        t,
+                        min(
+                            t + m.flows[i, o].nonconvex.minimum_downtime,
+                            m.TIMESTEPS.at(-1),
+                        ),
+                    )
                 )
                 return expr <= 0
             else:
-                expr = 0
-                expr += self.status[i, o, t]
-                expr += -m.flows[i, o].nonconvex.initial_status
-                return expr == 0
+                return Constraint.Skip
 
         return Constraint(
             self.MINDOWNTIMEFLOWS, m.TIMESTEPS, rule=min_downtime_rule
```

Comment:
Wouldn't something like
```suggestion
            self.MINDOWNTIMEFLOWS, m.TIMESTEPS[first_flexible_timestep:-1], rule=min_downtime_rule
```
work?

Reply:
m.TIMESTEPS[first_flexible_timestep:-1] does not work, because flexible_timestep  can variate for each element of the set MINDOWNTIMEFLOWS. Further m.TIMESTEPS is not a list (anymore), so the access has to done be in different way, e.g. [t for t in m.TIMESTEPS if t>y]

Reply:
You are right. We have a 2D array of constraints, so the current way might be the best one.